src/main/scala/viper/gobra/ast/frontend/Ast.scala

```diff
@@ -70,6 +70,7 @@ class PositionManager(val positions: Positions) extends Messaging(positions) {
     messages.sorted map { m =>
       errorFactory(
         formatMessage(m),
+        if (positions.getStart(m.value).isEmpty) None else 
```

Comment:
I added this line because it throws a better error. For example: (package `foo` imports package `bar`, but package `bar` has errors)
```
[info] Error at: Private members must all be defined in the private clause.
[info] Error at: <src\test\resources\regressions\features\private_spec\import\struct_import.gobra:7:8> src\test\resources\regressions\features\private_spec\import\struct_import.gobra:7:8:error: Package 'bar' contains errors: Vector(Private members must all be defined in the private clause.
```
If I remove the line `if (positions.getStart(m.value).isEmpty) None else`, it throws:
```
[info] An unknown Exception was thrown.
[info] None.get
[info] java.util.NoSuchElementException: None.get
[info]  at scala.None$.get(Option.scala:627)
[info]  at scala.None$.get(Option.scala:626)
[info]  at viper.gobra.ast.frontend.PositionManager.$anonfun$translate$1(Ast.scala:74)
[info]  at scala.collection.immutable.Vector1.map(Vector.scala:1872)
[info]  at scala.collection.immutable.Vector1.map(Vector.scala:375)
[info]  at viper.gobra.ast.frontend.PositionManager.translate(Ast.scala:70)
[info]  at viper.gobra.frontend.info.Info$.check(Info.scala:99)
[info]  at viper.gobra.frontend.info.implementation.resolution.MemberResolution.$anonfun$getTypeChecker$4(MemberResolution.scala:420)
[info]  at scala.util.Either.flatMap(Either.scala:352)
[info]  at viper.gobra.frontend.info.implementation.resolution.MemberResolution.$anonfun$getTypeChecker$3(MemberResolution.scala:416)
[info]  at scala.util.Either.flatMap(Either.scala:352)
[info]  at viper.gobra.frontend.info.implementation.resolution.MemberResolution.parseAndTypeCheck$1(MemberResolution.scala:413) 
[info]  at viper.gobra.frontend.info.implementation.resolution.MemberResolution.$anonfun$getTypeChecker$10(MemberResolution.scala:444)
[info]  at scala.Option.getOrElse(Option.scala:201)
[info]  at viper.gobra.frontend.info.implementation.resolution.MemberResolution.getTypeChecker(MemberResolution.scala:444)      
[info]  at viper.gobra.frontend.info.implementation.resolution.MemberResolution.getTypeChecker$(MemberResolution.scala:407)     
[info]  at viper.gobra.frontend.info.implementation.TypeInfoImpl.getTypeChecker(TypeInfoImpl.scala:22)
[info]  at viper.gobra.frontend.info.implementation.resolution.MemberResolution.tryPackageLookup(MemberResolution.scala:397)    
[info]  at viper.gobra.frontend.info.implementation.resolution.MemberResolution.tryPackageLookup$(MemberResolution.scala:395)   
[info]  at viper.gobra.frontend.info.implementation.TypeInfoImpl.tryPackageLookup(TypeInfoImpl.scala:22)
[info]  at viper.gobra.frontend.info.implementation.resolution.MemberResolution.tryDotLookup(MemberResolution.scala:326)        
[info]  at viper.gobra.frontend.info.implementation.resolution.MemberResolution.tryDotLookup$(MemberResolution.scala:296)       
[info]  at viper.gobra.frontend.info.implementation.TypeInfoImpl.tryDotLookup(TypeInfoImpl.scala:22)
[info]  at viper.gobra.frontend.info.implementation.resolution.NameResolution.$anonfun$entity$1(NameResolution.scala:303)       
[info]  at org.bitbucket.inkytonik.kiama.attribution.AttributionCore$CachedAttribute.liftedTree1$1(AttributionCore.scala:58)    
[info]  at org.bitbucket.inkytonik.kiama.attribution.AttributionCore$CachedAttribute.apply(AttributionCore.scala:56)
[info]  at viper.gobra.frontend.info.implementation.typing.IdTyping.$anonfun$wellDefID$1(IdTyping.scala:22)
...etc.
```
If I just try to verify package bar, it throws:
```
[info] Error at: <src\test\resources\regressions\features\private_spec\import\bar\bar.gobra:18:1> src\test\resources\regressions\features\private_spec\import\bar\bar.gobra:18:1:error: Private members must all be defined in the private clause.
```
I do not know if that is the intended way of getting the error message, or if I implemented it wrong, but None.get seems like a worse error message than simply saying that there is some error message in package `bar`.

Reply:
My point is that positions.getStart should never return None for parsed Nodes because we should guarantee that every parsed node has a position. Therefore, if it is none, then there is a bug somewhere else too. I suspect that in the Treetranslator, you are not adding positional data with `.at` for some node.